### PR TITLE
Topic/better recursive upload

### PR DESCRIPTION
**Pull request by erikd-ambiata**
Recursive upload now allows a fork parameter to specify the concurrency.
 
! @olorin @nhibberd

### CODE REVIEW

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -422,21 +422,27 @@ uploadWithMode m f a = do
...
-          multipartUpload f a s (2 * chunk) 20
+          multipartUpload f a s (2 * bigChunkSize) 20
         False ->
-          multipartUpload f a s chunk 20
+          multipartUpload f a s bigChunkSize 20
+
+
+
+bigChunkSize :: Integer
+bigChunkSize = 100 * 1024 * 1024
```
Comment (olorin) [inline mismi-s3/src/Mismi/S3/Commands.hs]: What's the reasoning behind this number?

**Comment by olorin** [comment]
This looks good to me.

👍

File: mismi-s3/src/Mismi/S3/Commands.hs (hunk 1 above)
Comment (nhibberd) [reply mismi-s3/src/Mismi/S3/Commands.hs]: Just extracted from the above function

Comment (erikd-ambiata) [reply mismi-s3/src/Mismi/S3/Commands.hs]: So it could be used elsewhere.

**Pull request merged**